sotodlib/preprocess/processes.py

```diff
@@ -585,6 +620,49 @@ class SubPolyf(_Preprocess):
     def process(self, aman, proc_aman):
         tod_ops.sub_polyf.subscan_polyfilter(aman, **self.process_cfgs)
 
+class SSOFootprint(_Preprocess):
+    """Find nearby sources within a given distance and get SSO footprint and plot
+    each source on the focal plane.
+
+    .. autofunction:: sotodlib.obs_ops.sources.get_sso
+    """
+    name = 'sso_footprint'
+
+    def calc_and_save(self, aman, proc_aman):
+        ssos = planets.get_nearby_sources(tod=aman, distance=self.calc_cfgs.get("distance", 20))
+        if ssos:
+            sso_aman = core.AxisManager()
+            for sso in ssos:
+                planet = sso[0]
+                xi_p, eta_p = obs_ops.sources.get_sso(aman, planet, nstep=self.calc_cfgs.get("nstep", 100))
+                planet_aman = core.AxisManager()
```

Comment:
This AxisManager should at least have a samps axis and when you wrap in xi_p/eta_p wrap them along the samps axis.

Reply:
The samps axis will be downsampled from `nstep` and this will downsample `proc_aman` too when wrap into it. Would this be an issue?

Reply:
You can create a new offset axis (called like `ds_samp`) and wrap them into that something like this:

```
planet_aman = core.AxisManager(core.OffsetAxis('ds_samps', count=len(downsampled_axis),
                                                          offset=aman.samps.offset, 
                                                          origin_tag=aman.samps.origin_tag)
```